### PR TITLE
4.5.0 Release

### PR DESCRIPTION
**Pull request by jennifer-shehane**
- 4.5.0 changelog
- updated docs on mocha custom reporter change #2701
- add component testing experimental config #2745

### CODE REVIEW

**Comment by bahmutov** [comment]
@jennifer-shehane we specifically moved the experimental page into its own guide, since it will be expanding soon with more info about component testing

**Comment by jennifer-shehane** [comment]
This needs to be reverted / moved out of deployment https://github.com/cypress-io/cypress-documentation/pull/2701

**Comment by jennifer-shehane** [comment]
Moved experiments back to its own page.

File: source/guides/references/experiments.md [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+```
+
+This test would execute in the browser, similar to the full end-to-end test, except there is no website URL:
+
+{% imgTag /img/guides/references/component-test.gif "Example React component test" %}
+
+Feature | Test libraries | Cypress component tests
+--- | --- | ---
+Test runs in real browser | ❌ | ✅
+Uses full mount | ❌ | ✅
```
Comment (JessicaSachs) [inline source/guides/references/experiments.md:42]: React Testing Library does a full mount. https://kentcdodds.com/blog/why-i-never-use-shallow-rendering

>  I've expressed this feeling on many occasions and get asked on a regular basis why I feel the way I do about shallow rendering and why React Testing Library will never support shallow rendering.

Comment (JessicaSachs) [reply source/guides/references/experiments.md:42]: Enzyme is the only one who doesn't do a full mount.. and that's optional.

Comment (brian-mann) [reply source/guides/references/experiments.md:42]: regardless of the mount - i don't think this is a real differentiator.

it's not so much about full or partial mount, or "whether it runs in a real browser"...

it's really about **seeing your components** and natively interacting with them. we should call to attention that you can see your components just like if you were in dev with a browser open pointing to `localhost`.

we should also call to attention videos and screenshots "just work" out of the box. no need to deal with an artificial DOM. it's the same DOM and the same interaction and rendering engine. there's no translation layer

**Pull request merged**